tools/distpartitioning/parmetis_preprocess.py

```diff
@@ -0,0 +1,326 @@
+    for ntype_id, ntype_name in ntid_ntype_map.items():
+        global_nid_offset = ntype_gnid_offset[ntype_name][0,0]
+        for r in range(num_parts):
+            type_start, type_end = type_nid_dict[ntype_name][r][0], type_nid_dict[ntype_name][r][1]
+            out_file = os.path.join(outdir, 'node_weights_{}_{}.txt'.format(ntype_name, r))
+            node_files.append((out_file, global_nid_offset + type_start, global_nid_offset + type_end))
+
+    nfile = open(os.path.join(params.output, 'parmetis_nfiles.txt'), "w")
+    for f in node_files:
+        # format: filename global_node_id_start global_node_id_end(exclusive)
+        nfile.write('{} {} {}\n'.format(f[0], f[1], f[2]))
+    nfile.close()
+
+    #Regenerate edge files here.
+    edge_data = schema_map[constants.STR_EDGES]
+    edge_files = []
+    for etype_name, etype_info in edge_data.items():
+        edge_info = etype_info[constants.STR_DATA]
+        for r in range(num_parts):
+            out_file = edge_info[r].split("/")[-1]
+            out_file = os.path.join(outdir, 'edges_{}'.format(out_file))
+            edge_files.append(out_file)
+
+    efile = open(os.path.join(params.output, 'parmetis_efiles.txt'), "w")
+    for f in edge_files:
+        efile.write('{}\n'.format(f))
+    efile.close()
+
+
+def run_preprocess_data(params):
+    """
+    Main function which will help create graph files for ParMETIS processing
+
+    Parameters:
+    -----------
+    params : argparser object
+	an instance of argparser class which stores command line arguments
+    """
+    logging.info(f'Starting to generate ParMETIS files...')
+
+    rank = get_proc_info()
+    schema = read_json(params.schema)
+    num_nodes_per_chunk = schema[constants.STR_NUM_NODES_PER_CHUNK]
+    num_parts = len(num_nodes_per_chunk[0])
+    gen_node_weights_files(schema, params.output)
+    logging.info(f'Done with node weights....')
+
+    gen_edge_files(schema, params.output)
+    logging.info(f'Done with edge weights...')
+
+    if rank == 0:
+        gen_parmetis_input_args(params, schema)
+    logging.info(f'Done generating files for ParMETIS run ..')
+
+if __name__ == "__main__":
+    """
+    Main function used to generate temporary files needed for ParMETIS execution.
+    This function generates node-weight files and edges files which are consumed by ParMETIS.
+
+    Example usage:
+    --------------
+    mpirun -np 4 python3 parmetis_preprocess.py --schema <file> --output <target-output-dir>
+    """
+    parser = argparse.ArgumentParser(description='Generate ParMETIS files for input dataset')
+    parser.add_argument('--schema', required=True, type=str,
+                     help='The schema of the input graph')
+    parser.add_argument('--output', required=True, type=str,
+                    help='The output directory for the node weights files and auxiliary files for ParMETIS.')
+    params = parser.parse_args()
+
+    #Invoke the function to generate files for parmetis
```

Comment:
white space after #.
https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4

You can use black to fix a lot of issues like this.
I also noticed that you mix both single quote ' and double quote " in the code.
Let's make it consistent.